dissect/target/plugins/scrape/qfind.py

```diff
@@ -102,68 +129,46 @@ def qfind(
             needle_lookup = tmp
 
+        elif regex:
+            tmp = {}
+            for needle, _ in needle_lookup.items():
+                tmp[re.compile(needle, re.IGNORECASE if ignore_case else re_NOFLAG)] = _
+            needle_lookup = tmp
+
         seen = set()
-        for _, stream, needle, offset in self.target.scrape.find(
-            list(needle_lookup.keys()), progress=progress(self.target)
+        for disk, stream, needle, offset, match in self.target.scrape.find(
+            list(needle_lookup.keys()), progress=progress
         ):
             original_needle, codec = needle_lookup[needle]
             needle_len = len(needle.pattern if isinstance(needle, re.Pattern) else needle)
             before_offset = max(0, offset - window)
             stream.seek(before_offset)
             buf = stream.read((offset - before_offset) + max(window, needle_len))
 
-            header = f"\r[{offset:#08x} @ {original_needle} ({codec})]"
+            if unique:
+                digest = hashlib.sha1(buf).digest() if window > 20 else buf
 
-            if raw:
-                print(header)
-                palette = [(offset - before_offset, utils.COLOR_NORMAL), (needle_len, utils.COLOR_BG_RED)]
-                utils.hexdump(buf, palette, offset=before_offset)
-            else:
-                codec = "utf-8" if codec == "hex" else codec
-                before_part = recover_string(
-                    buf[: offset - before_offset], codec, reverse=True, ascii=not allow_non_ascii
-                )
-                after_part = recover_string(buf[offset - before_offset :], codec, ascii=not allow_non_ascii)
-                hit = (
-                    before_part
-                    + utils.COLOR_BG_RED
-                    + after_part[:needle_len]
-                    + utils.COLOR_NORMAL
-                    + after_part[needle_len:]
-                )
-
-                if unique:
-                    if hit in seen:
-                        continue
-                    seen.add(hit)
-
-                print(header)
-                print(hit)
-
-
-def progress(target: Target) -> Callable[[Container | Volume, int, int], None]:
-    """Progress handler of the qfind plugin."""
-    current_disk = None
-
-    def update(disk: Container | Volume, offset: int, size: int) -> None:
-        nonlocal current_disk
-        if current_disk is None:
-            sys.stderr.write(f"{target}\n")
-
-        if current_disk != disk:
-            sys.stderr.write(f"[Current disk: {disk}]\n")
-            current_disk = disk
-
-        sys.stderr.write(f"\r{offset / float(size) * 100:0.2f}%")
-        sys.stderr.flush()
-
-    return update
+                if digest in seen:
+                    continue
+                seen.add(digest)
+
+            match = match.group() if match else original_needle
+
+            yield QFindMatchRecord(
+                disk=disk,
```

Comment:
```suggestion
                disk=repr(disk),
```
To make it a bit more explicit what's happening.